portalnet/src/storage.rs

```diff
@@ -744,6 +677,496 @@ impl PortalStorage {
+            BeaconContentKey::from_ssz_bytes(content_key.as_slice()).map_err(|err| {
+                ContentStoreError::InvalidData {
+                    message: format!("Error deserializing BeaconContentKey value: {err:?}"),
+                }
+            })?;
+
+        match beacon_content_key {
+            BeaconContentKey::LightClientBootstrap(_) => {
+                let content_id = key.content_id();
+                self.lookup_content_value(content_id).map_err(|err| {
+                    ContentStoreError::Database(format!(
+                        "Error looking up LightClientBootstrap content value: {err:?}"
+                    ))
+                })
+            }
+            BeaconContentKey::LightClientUpdatesByRange(content_key) => {
+                let periods =
+                    content_key.start_period..(content_key.start_period + content_key.count);
+
+                let mut content: Vec<ForkVersionedLightClientUpdate> = Vec::new();
+
+                for period in periods {
+                    let result = self.lookup_lc_update_value(period).map_err(|err| {
+                        ContentStoreError::Database(format!(
+                            "Error looking up LightClientUpdate content value: {err:?}"
+                        ))
+                    })?;
+
+                    match result {
+                        Some(result) => content.push(
+                            ForkVersionedLightClientUpdate::from_ssz_bytes(result.as_slice())
+                                .map_err(|err| {
+                                    ContentStoreError::Database(format!(
+                                    "Error ssz decode ForkVersionedLightClientUpdate value: {err:?}"
+                                ))
+                                })?,
+                        ),
+                        None => return Ok(None),
+                    }
+                }
+
+                let result = VariableList::<ForkVersionedLightClientUpdate, U128>::new(content)
+                    .map_err(|err| ContentStoreError::Database(
+                        format!(
+                            "Error building VariableList from ForkVersionedLightClientUpdate data: {err:?}"
+                        ),
+                    ))?;
+
+                Ok(Some(result.as_ssz_bytes()))
+            }
+            BeaconContentKey::LightClientFinalityUpdate(_) => {
+                // TODO: Get LightClientFinalityUpdate from cache
+                Ok(None)
+            }
+            BeaconContentKey::LightClientOptimisticUpdate(_) => {
+                // TODO: Get LightClientOptimisticUpdate from cache
+                Ok(None)
+            }
+        }
+    }
+
+    fn put<K: OverlayContentKey, V: AsRef<[u8]>>(
+        &mut self,
+        key: K,
+        value: V,
+    ) -> Result<(), ContentStoreError> {
+        self.store(&key, &value.as_ref().to_vec())
+    }
+
+    /// The "radius: concept is not applicable for Beacon network
```

Comment:
Seems like this function name/purpose should be updated in upcoming refactoring so that it can be shared amongst storages while also being logically consistent.